mesonbuild/interpreter/interpreter.py

```diff
@@ -2638,6 +2639,16 @@ def func_get_variable(self, node: mparser.BaseNode, args: T.Tuple[T.Union[str, D
     def func_is_variable(self, node: mparser.BaseNode, args: T.Tuple[str], kwargs: 'TYPE_kwargs') -> bool:
         return args[0] in self.variables
 
+    @typed_pos_args('unset_variable', str)
+    @noKwargs
+    @noArgsFlattening
+    def func_unset_variable(self, node: mparser.BaseNode, args: T.Tuple[str], kwargs: 'TYPE_kwargs') -> None:
+        varname = args[0]
+        if varname not in self.variables:
+            raise InterpreterException(f'Tried to unset unknown variable "{varname}".')
+
+        del self.variables[varname]
+
```

Comment:
I think it would be better to do this with a try/except. That will make the success path faster, and we don't care in the failure case, because that's fatal